pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/LoggingFilter.java

```diff
@@ -24,25 +25,30 @@
     public static final String USER_EMAIL = "USER_EMAIL";
 
     private static final Logger logger = LoggerFactory.getLogger(LoggingFilter.class);
+    private String auth0Domain;
+    private String claimNameSpace;
+
+    public LoggingFilter(String auth0Domain, String claimNameSpace){
+        this.auth0Domain = auth0Domain;
+        this.claimNameSpace = claimNameSpace;
+    }
 
     @Override
     public void handle(Request request, Response response) {
         String tokenFromHeader = Utility.getTokenFromHeader(request);
-        if (StringUtils.isNotBlank(tokenFromHeader)) {
-            try {
-                DecodedJWT decodedUnverifiedJWT = JWT.decode(tokenFromHeader);
-                Claim userEmailClaim = decodedUnverifiedJWT.getClaim("USER_MAIL");
-
-                if (userEmailClaim != null) {
-                    String userEmail = userEmailClaim.asString();
-                    if (StringUtils.isNotBlank(userEmail)) {
-                        MDC.put(USER_EMAIL, userEmail);
+        if (StringUtils.isNotBlank(tokenFromHeader) && !"null".equals(tokenFromHeader)) {
+                Optional<DecodedJWT> maybeDecodedUnverifiedJWT = JWTConverter.verifyDDPToken(tokenFromHeader, auth0Domain);
+                maybeDecodedUnverifiedJWT.ifPresentOrElse(decodedJWT -> {
+                    Claim userEmailClaim = decodedJWT.getClaim(claimNameSpace+"USER_MAIL");
+
+                    if (userEmailClaim != null) {
+                        String userEmail = userEmailClaim.asString();
+                        if (StringUtils.isNotBlank(userEmail)) {
+                            MDC.put(USER_EMAIL, userEmail);
+                        }
                     }
-                }
+                }, () ->{throw  new RuntimeException("Unable to verifiy token");});
```

Comment:
Is intent to block the request from proceeding to execute a route?
If so, proper way to do this can be found here:
https://github.com/broadinstitute/ddp-study-server/blob/1ef16842cc27f6d286c1ec8cfc7558b4b7452d75/pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/filter/StudyAdminAuthFilter.java#L29

Reply:
Thank you!! changes applied!